tests/test_pilot.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+
+from faker import Faker
+from game.squadrons.pilot import Pilot, PilotStatus
+
+
+@pytest.fixture(scope="function")
+def pilot():
+    return Pilot("John Doe")
+
+
+def test_pilot_creation() -> None:
+    new_pilot = Pilot("John Doe")
+    assert new_pilot.name == "John Doe"
+    assert new_pilot.status == PilotStatus.Active
+
+
+def test_pilot_active(pilot) -> None:
+    assert pilot.status == PilotStatus.Active
+
+
+def test_pilot_alive(pilot) -> None:
+    assert pilot.alive == True
+
+
+def test_pilot_on_leave(pilot) -> None:
+    pilot.send_on_leave()
+    assert pilot.status == PilotStatus.OnLeave
+    assert pilot.on_leave == True
+    pilot.return_from_leave()
+    assert pilot.status == PilotStatus.Active
+    assert pilot.on_leave == False
+
+
+def test_pilot_on_leave_twice(pilot) -> None:
+    pilot.send_on_leave()
+    assert pilot.status == PilotStatus.OnLeave
+    with pytest.raises(RuntimeError):
+        pilot.send_on_leave()
+
+
+def test_pilot_not_on_leave(pilot) -> None:
+    with pytest.raises(RuntimeError):
+        pilot.return_from_leave()
+
+
+def test_pilot_dead(pilot) -> None:
+    pilot.kill()
+    assert pilot.status == PilotStatus.Dead
+
+
+def test_pilot_record(pilot) -> None:
+    pilot.record.missions_flown == 0
+
+
+def test_missions_flown(pilot) -> None:
+    pilot.record.missions_flown = 1
+    assert pilot.record.missions_flown == 1
+
+
+def test_random_pilot_name() -> None:
+    faker = Faker()
+    random_pilot = Pilot.random(faker)
+    assert len(random_pilot.name) > 3
```

Comment:
As someone with a three letter name, I don't like this particular test ;)

It's not really safe to assume anything about a randomly generated name (https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/ is a fun read). `assert random_pilot.name` is good enough.

```suggestion
    assert random_pilot.name
```